ndk/src/thermal.rs

```diff
@@ -0,0 +1,331 @@
+//! Bindings for [`AThermalManager`]
+//!
+//! Structures and functions to access thermal status and register/unregister thermal status
+//! listener in native code.
+//!
+//! [`AThermalManager`]: https://developer.android.com/ndk/reference/group/thermal#athermalmanager
+#![cfg(feature = "api-level-30")]
+
+#[cfg(doc)]
+use std::io::ErrorKind;
+use std::{io::Result, os::raw::c_void, ptr::NonNull};
+
+use num_enum::{FromPrimitive, IntoPrimitive};
+
+use crate::utils::abort_on_panic;
+
+/// Workaround for <https://issuetracker.google.com/issues/358664965>.  `status_t` should only
+/// contain negative error codes, but the underlying `AThermal` implementation freely passes
+/// positive error codes around. At least the expected return codes are "implicitly" documented to
+/// be positive.
+fn status_to_io_result(status: i32) -> Result<()> {
+    // Intentionally not imported in scope (and an identically-named function) to prevent
+    // accidentally calling this function without negation.
+    crate::utils::status_to_io_result(-status)
+}
```

Comment:
Unlike all other NDK APIs, this one returns _positive_ errno values :upside_down_face: 

https://issuetracker.google.com/issues/358664965